testcases/crypto/aes_func.c

```diff
@@ -1915,6 +1929,298 @@ CK_RV do_SignVerifyMAC(struct published_mac_test_suite_info *tsuite)
     return rc;
 }
 
+CK_RV do_SetAttributeValuesPkey()
+{
+    CK_BYTE user_pin[PKCS11_MAX_PIN_LEN];
+    CK_FLAGS flags;
+    CK_ULONG user_pin_len;
+    CK_SLOT_ID slot_id = SLOT_ID;
+
+    CK_OBJECT_HANDLE h_key = CK_INVALID_HANDLE;
+    CK_SESSION_HANDLE session;
+    CK_MECHANISM keygen_mech = { CKM_AES_KEY_GEN, 0, 0 };
+    CK_RV rc;
+
+    CK_BBOOL pkey_extr;
+    CK_ATTRIBUTE tmpl[] = {
+        {CKA_IBM_PROTKEY_EXTRACTABLE, &pkey_extr, sizeof(CK_BBOOL)},
+    };
+    CK_ULONG tmpl_len = sizeof(tmpl) / sizeof(CK_ATTRIBUTE);
+
+
+    testsuite_begin("do_SetAttributeValuesPkey");
+    testcase_rw_session();
+    testcase_user_login();
+
+    /* Skip tests if the slot doesn't support protected keys*/
+    if (!is_ep11_token(SLOT_ID)) {
+        testsuite_skip(1, "Slot %u doesn't support protected keys",
+                       (unsigned int) slot_id);
+        goto testcase_cleanup;
+    }
+
+    testcase_begin("Generate AES key and change CKA_IBM_PROTKEY_EXTRACTABLE");
+    testcase_new_assertion();
+
+    /* Generate a test key */
+    rc = generate_AESKey(session, 16, CK_FALSE, &keygen_mech, &h_key);
+    if (rc != CKR_OK) {
+        testcase_fail("generate_AESKey rc=%s", p11_get_ckr(rc));
```

Comment:
> Thinking about it, this is actually the correct place to fail the test and `generate_AESKey` should not fail the test. Instead, it should just return a failure code.

Yes, but generate_AESKey is used that way in many testcases, so changing this would also require to change all of them.